service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java

```diff
@@ -119,6 +121,14 @@ public void handleSubmittedEvent(
         if (featureToggleService.isPaymentsEnabled()) {
             paymentService.makePaymentForC2(caseDetails.getId(), caseData);
         }
+
+        C2DocumentBundle c2DocumentBundle = caseData.getLastC2DocumentBundle();
+
+        if (NO.getValue().equals(c2DocumentBundle.getUsePbaPayment())) {
```

Comment:
Would it make sense to extract that bit to private method?

```
public boolean isPaidByPba(CaseData caseData) {
    return NO.getValue().equals(caseData.getLastC2DocumentBundle().getUsePbaPayment());
}
```

Reply:
Maybe not pass caseData that seems a bit to much for this method